crates/rpc/engine/exchange_transition_config.rs

```diff
@@ -0,0 +1,78 @@
+use ethereum_rust_core::{serde_utils, H256};
+use ethereum_rust_storage::Store;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use tracing::{info, warn};
+
+use crate::{utils::RpcErr, RpcHandler};
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExchangeTransitionConfigPayload {
+    #[serde(with = "serde_utils::u128::hex_str")]
+    terminal_total_difficulty: u128,
+    terminal_block_hash: H256,
+    #[serde(with = "serde_utils::u64::hex_str")]
+    terminal_block_number: u64,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExchangeTransitionConfigV1Req {
+    payload: ExchangeTransitionConfigPayload,
+}
+
+impl std::fmt::Display for ExchangeTransitionConfigV1Req {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "ExchangeTransitionConfigV1Req {{ terminal_total_difficulty: {}, terminal_block_hash: {:?}, terminal_block_number: {} }}",
+            self.payload.terminal_total_difficulty,
+            self.payload.terminal_block_hash,
+            self.payload.terminal_block_number
+        )
+    }
+}
+
+impl RpcHandler for ExchangeTransitionConfigV1Req {
+    fn parse(params: &Option<Vec<Value>>) -> Option<ExchangeTransitionConfigV1Req> {
+        let params = params.as_ref()?;
+        if params.len() != 1 {
+            return None;
+        };
+        let params: ExchangeTransitionConfigPayload =
+            serde_json::from_value(params[0].clone()).unwrap();
+        Some(ExchangeTransitionConfigV1Req {
+            payload: ExchangeTransitionConfigPayload {
+                terminal_total_difficulty: params.terminal_total_difficulty,
+                terminal_block_hash: params.terminal_block_hash,
+                terminal_block_number: params.terminal_block_number,
+            },
+        })
+    }
+
+    fn handle(&self, storage: Store) -> Result<Value, RpcErr> {
+        info!("Received new engine request: {self}");
```

Comment:
you also need to validate here that you're in a fork that is previous to cancun since this has been deprecated since cancun

Reply:
Sorry, I actually misread the spec: https://github.com/ethereum/execution-apis/blob/main/src/engine/cancun.md#deprecate-engine_exchangetransitionconfigurationv1

I think the deprecation notice is not really needed after all :/

Reply:
No problem, I'll remove it!